### PR TITLE
Implement Thrift backend

### PR DESCRIPTION
**Pull request by dshiga**
Based on Hussein's thrift_backend branch. That branch made most of the changes needed to use thrift, and allowed for submitting jobs via thrift - but checking job status with thrift was not fully implemented. The changes in this PR finish implementing the job status call via thrift.

### CODE REVIEW

File: herc/webservice.py [hunk 1]
```diff
@@ -22,6 +22,9 @@ def initialize(self):
         self.set_header('Content-Type', 'application/json')
         self.set_header('Access-Control-Allow-Origin',	   '*')
         self.set_header('Access-Control-Allow-Credentials', 'true')
+        module_dir = os.path.dirname(__file__)
```
Comment (helgridly) [inline herc/webservice.py]: Hmm. If we're going to do this, we should do it for all referenced data files, and do it something like [this question](http://stackoverflow.com/a/5423147/2941784).
@scottfrazer Thoughts?

File: setup.py [hunk 2]
```diff
@@ -12,10 +14,8 @@ def readme():
       url='http://github.com/broadinstitute/herc',
       author='The Broad Institute',
       packages=find_packages(exclude='tests'),
-      package_data={
-          # Include everything in data/, both schemas and examples.
-          '': ['data/*']
-      },
+      data_files=[('data/aurora', ['data/aurora/api.thrift']),
+                  ('data/schemas', ['data/schemas/jobsubmit.json'])],
```
Comment (helgridly) [inline setup.py]: What's the reason for this change? Was `data/*` not working before?

File: herc/webservice.py [hunk 3]
```diff
@@ -22,6 +22,9 @@ def initialize(self):
...
+        self.submit_schema = os.path.join(module_dir, os.pardir, 'data', 'schemas', 'jobsubmit.json')
+        print(self.submit_schema)
```
Comment (scottfrazer) [inline herc/webservice.py]: This print statement probably shouldn't be here.

File: setup.py (hunk 2 above)
Comment (dshiga) [inline setup.py]: Correct - nothing in the data dir was getting copied.

File: herc/webservice.py (hunk 1 above)
Comment (scottfrazer) [inline herc/webservice.py]: Yeah that seems like a good approach to me

File: herc/webservice.py [hunk 4]
```diff
@@ -77,7 +80,7 @@ class schema(base):
     def get(self):
         """GET /schema
         Returns the JSON schema used to validate job submission requests."""
-        with open("data/schemas/jobsubmit.json", 'r', encoding="utf-8") as jschema:
+        with open(self.submit_schema, 'r', encoding="utf-8") as jschema:
```
Comment (scottfrazer) [inline herc/webservice.py]: So, if we're going with @helgridly's suggestion, then this could be something like this for clarity:

```python
with open(get_data_path('data/schema/jobsubmit.json'), 'r', encoding="utf-8") as jschema:
```

File: setup.py (hunk 2 above)
Comment (helgridly) [inline setup.py]: That's upsetting. There's more in the data/ dir that should be packaged, though. I suppose you could specify everything manually but something in [this big mess of words](https://pythonhosted.org/setuptools/setuptools.html#including-data-files) should be what you need.

File: setup.py [hunk 5]
```diff
@@ -25,9 +25,13 @@ def readme():
...
           'mock',
           'arrow',
-          'nose'
+          'nose',
+          'requests',
+          'thrift==1.0.0-py3',
+          'munch'
       ],
       entry_points={'console_scripts': ['herc = herc.webservice:main']},
+      dependency_links = [urllib.parse.urljoin('file:', os.path.join(os.getcwd(), 'thrift-1.0.0-py3.tar.gz'))],
```
Comment (scottfrazer) [inline setup.py]: What's this all about?

Comment (helgridly) [inline setup.py]: That's my terrible doing. I added a tarball of my Python3 build of the Thrift library to the repo; this is some magic to reference that when attempting to satisfy the Thrift dependency.

It should be added to the package files if it isn't already.

Comment (scottfrazer) [inline setup.py]: I'm not sure I understand why it's needed.  Does it get untarred somehow?

Comment (helgridly) [inline setup.py]: What's happening here is a version of [this](http://stackoverflow.com/a/3481388/2941784): specify a version that doesn't exist on PyPI, then satisfy that version yourself in dependency_links. The path joining is to use a local file as an install source, rather than specifying e.g. the GitHub tarball (which is more common).

I suspect this means the tarball should be added to `package_data` or `data_files` or whatever ends up being used (see above comments).

Comment (dshiga) [inline setup.py]: That tarball is already getting copied to the appropriate place - is there another reason why we should add it to package_data?

Comment (helgridly) [inline setup.py]: In the event that this project gets pushed to PyPI, it might only upload data explicitly referenced in the setup.py. But honestly I have no idea. I do know it works off a fresh `git clone`, so we could potentially just punt the issue for now.

File: herc/webservice.py (hunk 4 above)
Comment (dshiga) [inline herc/webservice.py]: Ok I made that change.

**Comment by dshiga** [comment]
Added the thrift tarball to data files in setup.py. I couldn't figure out how to properly include data files without specifying them one by one -- the methods described in the docs don't seem to work consistently.
